Source/Models/APIClient.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+import Argo
+import Result
+
+public struct APIClient {
+  private let requestPerformer: RequestPerformer
+
+  public init(requestPerformer: RequestPerformer = NetworkRequestPerformer()) {
+    self.requestPerformer = requestPerformer
+  }
+}
+
+public extension APIClient {
+  func performRequest<T: Request>(request: T, completionHandler: Result<T.ResponseType, NSError> -> Void) {
+    requestPerformer.performRequest(request.build()) { result in
+      let object = result >>- deserialize >>- { request.parse($0) }
+      completionHandler(object)
+    }
+  }
+}
+
+private func deserialize(response: HTTPResponse) -> Result<JSON, NSError> {
+  do {
+    let data = response.data ?? NSData()
```

Comment:
That makes sense. I suppose in this context we're expecting it to return a JSON response to be parsed, so if it doesn't, then something is wrong. 

I agree it seems reasonable that an `HTTPResponse` could not have data associated(204 No Content) but in this context it's a failure. 

All that being said, did you consider providing a descriptive error message stating that there was no content and we expected there to be? 

I'm not sure how helpful a failed json parsing error would be to someone. Especially since we know exactly what happened we can give a pretty message to help the user out.